test/JavaEchoServerTest.java

```diff
@@ -1,31 +1,42 @@
-import org.junit.After;
-import org.junit.Before;
 import org.junit.Test;
 
-import java.io.ByteArrayOutputStream;
-import java.io.PrintStream;
+import java.io.*;
 
 import static org.junit.Assert.assertEquals;
 
 public class JavaEchoServerTest {
 
-    private ByteArrayOutputStream stdOut;
+    @Test
+    public void endToEnd() throws IOException {
```

Comment:
When I run this test locally it fails 😿 It seems a bit of a complicated setup. Is there a way that you can create an instance of `JavaEchoServer` and invoke the `echoInput` method on it directly?  You may need to think about passing some parameters into the method, which might also help simplify this test.